### PR TITLE
Use XtalClock as UART source and adapt interrupt binding

### PR DESCRIPTION
**Pull request by SergioGasquez**
- Use `XtalClock` as UART source when possible
- Update interrupt bindings

## Testing
I've tried flashing a few targets (C6 and H2) with the generated stubs and they were working fine. 

I've created https://github.com/SergioGasquez/espflash/tree/feat/test-stubs updating all the stubs ~and once https://github.com/esp-rs/espflash/pull/625 is merged, I'll run espflash HIL CI on that branch so we can test all the targets with new stubs.~

Here is the HIL run with the test stubs: https://github.com/esp-rs/espflash/actions/runs/8749962162

### CODE REVIEW

File: src/io/usb_serial_jtag.rs [hunk 1]
```diff
@@ -42,5 +43,5 @@ unsafe fn USB_DEVICE() {
 
     reg_block
         .int_clr()
-        .write(|w| w.serial_out_recv_pkt_int_clr().set_bit());
+        .write(|w| w.serial_out_recv_pkt().clear_bit_by_one());
```
Comment (SergioGasquez) [inline src/io/usb_serial_jtag.rs:46]: https://github.com/esp-rs/esp-hal/pull/1357/files#diff-2f29dbf7709dea452a2a05553431619b985294c24456e4d8ffea46e13babfc89L308

File: src/io/uart.rs [hunk 2]
```diff
@@ -41,5 +46,5 @@ fn UART0() {
         unsafe { RX_QUEUE.push_back(data).unwrap() };
     }
 
-    uart.int_clr().write(|w| w.rxfifo_full_int_clr().set_bit());
+    uart.int_clr().write(|w| w.rxfifo_full().clear_bit_by_one());
```
Comment (SergioGasquez) [inline src/io/uart.rs:49]: https://github.com/esp-rs/esp-hal/pull/1357/files#diff-44c26a357c3370ef8d7c4748ff6930fd0b4c6f75d705ce823440a42687c358d1L797

File: Cargo.toml [hunk 3]
```diff
@@ -8,8 +8,8 @@ categories   = ["embedded", "no-std"]
 
 [dependencies]
 critical-section = "1.1.2"
-esp-hal = { version = "0.16.0" }
-heapless         = { version = "0.8.0",  default-features = false }
+esp-hal = { git = "https://github.com/esp-rs/esp-hal", rev = "8f2ee88" }
```
Comment (SergioGasquez) [inline Cargo.toml:11]: To be updated when `esp-hal` is released:
```suggestion
esp-hal          = "0.17.0"
```

File: src/io/uart.rs [hunk 4]
```diff
@@ -20,10 +25,10 @@ where
     }
 }
 
-impl<T> UartMarker for Uart<'_, T> where T: Instance {}
+impl<T> UartMarker for Uart<'_, T, Blocking> where T: Instance {}
 
-#[interrupt]
-fn UART0() {
+#[handler]
+pub fn uart0_hanlder() {
```
Comment (JurajSadel) [inline src/io/uart.rs:31]: Typo: hanlder

```suggestion
pub fn uart0_handler() {
```

File: src/main.rs [hunk 5]
```diff
@@ -97,25 +93,45 @@ fn main() -> ! {
...
+        #[cfg(any(feature = "esp32", feature = "esp32s2"))]
+        clock_source: ClockSource::Apb,
+    };
+
+    let mut serial = Uart::new_with_config(
+        uart0,
+        uart_config,
+        None::<TxRxPins<gpio::NoPinType, gpio::NoPinType>>,
+        clocks,
+        Some(flasher_stub::io::uart::uart0_hanlder),
```
Comment (JurajSadel) [inline src/main.rs:112]: same typo:

```suggestion
        Some(flasher_stub::io::uart::uart0_handler),
```

**Review by JurajSadel — COMMENTED** [review]
Mostly LGTM, just one typo in the interrupt handler

**Review by MabezDev — APPROVED** [review]
LGTM, thank you!

**Pull request merged**